### PR TITLE
Incrementing the version number to `0.20.0-dev`

### PR DESCRIPTION
**Pull request by antalszava**
Updates PennyLane and PennyLane-QChem as we're entering a new development version (v0.19.0 coming up):

* Moves the v0.19.0 changelog into a separate file
* Updates the dev changelog file to have empty sections
* Increments the version number

### CODE REVIEW

File: doc/releases/changelog-0.19.0.md [hunk 1]
```diff
@@ -0,0 +1,1083 @@
+:orphan:
+
+# Release 0.19.0-dev (current release)
```
Comment (rmoyard) [inline doc/releases/changelog-0.19.0.md:3]:
```suggestion
# Release 0.19.0 (current release)
```

**Review by rmoyard — COMMENTED** [review]
Looks good to me 💯 just one small mistake I think

**Review by rmoyard — APPROVED** [review]
Perfect 💯

**Review by josh146 — APPROVED** [review]
Thanks Antal!

**Pull request merged**